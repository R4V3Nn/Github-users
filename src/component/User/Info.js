import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchUser } from '../../actions/userActions';
import Spinner from '../Spinner';

import './user-info.scss';

class UserInfo extends Component {
	componentDidMount() {
		this.props.loadUser(7);
	}
	render() {
		const { user } = this.props;
		const fields = ['login', 'name', 'company', 'blog', 'email', 'followers'];

		if (!user.login) {
			return <Spinner />;
		}

		return (
			<div className="user">
				<img className="image" src={user.avatar_url} alt="avatar" />
				<ul className="info">
					{
						fields.map(field => (
							<li key={field}>
								<span>{`${field}: `}</span>
								{user[field]}
							</li>
						))
					}

				</ul>

			</div>
		);
	}
}

UserInfo.propTypes = {
	user: PropTypes.object.isRequared,
	loadUser: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
	user: state.users.user,
});

const mapDispatchToProps = dispatch => ({
	loadUser: id => dispatch(fetchUser(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(UserInfo);

