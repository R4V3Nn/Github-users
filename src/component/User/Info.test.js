import React from 'react';
import { mount } from 'enzyme';

import { UserInfo } from './Info';

describe('Info component', () => {
	const props = {
		user: {
			login: '10',
			id: 21185613,
			node_id: 'MDEyOk9yZ2FuaXphdGlvbjIxMTg1NjEz',
			avatar_url: 'https://avatars2.githubusercontent.com/u/21185613?v=4',
			gravatar_id: '',
			url: 'https://api.github.com/users/10',
			html_url: 'https://github.com/10',
			followers_url: 'https://api.github.com/users/10/followers',
			following_url: 'https://api.github.com/users/10/following{/other_user}',
			gists_url: 'https://api.github.com/users/10/gists{/gist_id}',
			starred_url: 'https://api.github.com/users/10/starred{/owner}{/repo}',
			subscriptions_url: 'https://api.github.com/users/10/subscriptions',
			organizations_url: 'https://api.github.com/users/10/orgs',
			repos_url: 'https://api.github.com/users/10/repos',
			events_url: 'https://api.github.com/users/10/events{/privacy}',
			received_events_url: 'https://api.github.com/users/10/received_events',
			type: 'Organization',
			site_admin: false,
			name: null,
			company: null,
			blog: '',
			location: null,
			email: null,
			hireable: null,
			bio: null,
			public_repos: 0,
			public_gists: 0,
			followers: 0,
			following: 0,
			created_at: '2016-08-23T00:32:30Z',
			updated_at: '2019-07-17T16:28:11Z',
		},
		loadUser: jest.fn(),
	};

	it('should render UserInfo component', () => {
		// when
		const wrapper = mount(<UserInfo {...props} />);

		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
