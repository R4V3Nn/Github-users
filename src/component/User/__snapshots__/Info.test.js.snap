// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info component should render UserInfo component 1`] = `
<UserInfo
  loadUser={
    [MockFunction] {
      "calls": Array [
        Array [
          7,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  user={
    Object {
      "avatar_url": "https://avatars2.githubusercontent.com/u/21185613?v=4",
      "bio": null,
      "blog": "",
      "company": null,
      "created_at": "2016-08-23T00:32:30Z",
      "email": null,
      "events_url": "https://api.github.com/users/10/events{/privacy}",
      "followers": 0,
      "followers_url": "https://api.github.com/users/10/followers",
      "following": 0,
      "following_url": "https://api.github.com/users/10/following{/other_user}",
      "gists_url": "https://api.github.com/users/10/gists{/gist_id}",
      "gravatar_id": "",
      "hireable": null,
      "html_url": "https://github.com/10",
      "id": 21185613,
      "location": null,
      "login": "10",
      "name": null,
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjIxMTg1NjEz",
      "organizations_url": "https://api.github.com/users/10/orgs",
      "public_gists": 0,
      "public_repos": 0,
      "received_events_url": "https://api.github.com/users/10/received_events",
      "repos_url": "https://api.github.com/users/10/repos",
      "site_admin": false,
      "starred_url": "https://api.github.com/users/10/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/10/subscriptions",
      "type": "Organization",
      "updated_at": "2019-07-17T16:28:11Z",
      "url": "https://api.github.com/users/10",
    }
  }
/>
`;
